dagshub/data_engine/model/query_result.py

```diff
@@ -956,7 +963,7 @@ def generate_predictions(
                 _Batcher(dset, batch_size) if batch_size != 1 else dset
             ):  # encapsulates dataset with batcher if necessary and iterates over it
                 for prediction, remote_path in zip(
-                    predict_fn(local_paths),
+                    [predict_fn(local_paths)],
                     [result.path for result in self[idx * batch_size : (idx + 1) * batch_size]],
```

Comment:
[blocker] This doesn't look right.
The signature of `CustomPredictor` is:
```python
CustomPredictor = Callable[
    [List[str]],
    List[Tuple[Any, Optional[float]]],
]
```

So the function should be accepting a list of datapoints, and return a list of prediction tuples. 
Why are you wrapping the result in another list? This zip will end up processing all of the returned predictions for one single path.

Reply:
How does the return from the prediction function you're using right now look like?

Reply:
Apparently this is happening only when batch size = 1. Bigger batch sizes are supposed to work correctly.
Honestly don't know why not just use the Batcher even for batch size = 1, but I don't want to introduce too many changing variables right now, considering I can't test it that well.